fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -98,25 +99,80 @@ public MaterializedViewRewriter(MaterializationContext materializationContext) {
         this.materializationContext = materializationContext;
     }
 
+    /**
+     * Whether the MV expression can be satisfiable for input query, eg AggregateScanRule require
+     * MV expression should contain Aggregation operator.
+     *
+     * @param expression: MV expression.
+     */
     public boolean isValidPlan(OptExpression expression) {
         return MvUtils.isLogicalSPJ(expression);
     }
 
+    private boolean isMVApplicable(OptExpression queryExpression,
+                                   OptExpression mvExpression,
+                                   List<Table> queryTables,
+                                   List<Table> mvTables) {
+        if (!isValidPlan(mvExpression)) {
+            return false;
+        }
+
+        // If table lists do not intersect, can not be rewritten
+        if (Collections.disjoint(queryTables, mvTables)) {
+            return false;
+        }
+
+        // Only care MatchMode.COMPLETE here, QUERY_DELTA also can be supported
+        // because optimizer will match MV's pattern which is subset of query opt tree
+        // from top-down iteration.
+        MatchMode matchMode = getMatchMode(queryTables, mvTables);
+        if (matchMode != MatchMode.COMPLETE) {
+            return false;
+        }
+
+        boolean isQueryAllEqualInnerJoin = MvUtils.isAllEqualInnerJoin(queryExpression);
+        boolean isMVAllEqualInnerJoin = MvUtils.isAllEqualInnerJoin(mvExpression);
+        if (isQueryAllEqualInnerJoin && isMVAllEqualInnerJoin) {
+            return true;
+        } else {
+            // If not all join types are InnerJoin, need to check whether MV's join tables' order
+            // matches Query's join tables' order.
+            // eg. MV   : a left join b inner join c
+            //     Query: b left join a inner join c (cannot rewrite)
+            //     Query: a left join b inner join c (can rewrite)
+            //     Query: c inner join a left join b (can rewrite)
```

Comment:
what if the following case?
mv: a left join b inner join c inner join d
query: a left join b inner join d inner join c

Reply:
need reply JoinReorder's rule to match all join tables' order is the same, so can be matched with MV later.